### PR TITLE
Refactor: move azure config to config package

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/cache/azure_cache.go [hunk 1]
```diff
@@ -88,8 +88,8 @@ type ResourceProvider struct {
 	Getter GetFunc
 }
 
-// NewTimedCache creates a new azcache.Resource.
-func NewTimedCache(ttl time.Duration, getter GetFunc, disabled bool) (Resource, error) {
+// NewCachedResourceRepo creates a new azcache.Resource.
+func NewCachedResourceRepo(ttl time.Duration, getter GetFunc, disabled bool) (Resource, error) {
```
Comment (feiskyer) [inline pkg/cache/azure_cache.go:92]: any reasons to rename this function? and what does 'Repo' mean here?

File: pkg/provider/azure_loadbalancer_repo_test.go [hunk 2]
```diff
@@ -54,7 +55,7 @@ func TestListManagedLBs(t *testing.T) {
 	tests := []struct {
 		existingLBs, expectedLBs []network.LoadBalancer
 		callTimes                int
-		multiSLBConfigs          []MultipleStandardLoadBalancerConfiguration
+		multiSLBConfigs          []ratelimitconfig.MultipleStandardLoadBalancerConfiguration
```
Comment (jwtty) [inline pkg/provider/azure_loadbalancer_repo_test.go:58]: In azure_loadbalancer.go, this is **azureconfig**.MultipleStandard....

File: pkg/provider/azure_loadbalancer_test.go [hunk 3]
```diff
@@ -7139,33 +7141,33 @@ func TestGetEligibleLoadBalancers(t *testing.T) {
 		{
 			description: "should respect service annotation",
 			svc:         getTestService("test", v1.ProtocolTCP, map[string]string{consts.ServiceAnnotationLoadBalancerConfigurations: "A ,b"}, false),
-			lbConfigs: []MultipleStandardLoadBalancerConfiguration{
+			lbConfigs: []ratelimitconfig.MultipleStandardLoadBalancerConfiguration{
```
Comment (jwtty) [inline pkg/provider/azure_loadbalancer_test.go:7144]: Why do we use ratelimitconfig. and azureconfig. at the same time?

Comment (MartinForReal) [reply pkg/provider/azure_loadbalancer_test.go:7144]: I will rename it to azureconfig

File: pkg/cache/azure_cache.go (hunk 1 above)
Comment (MartinForReal) [reply pkg/cache/azure_cache.go:92]: repo is short for repository. I will raise another pr and replace timedcache with the one in client-go

File: pkg/provider/azure_loadbalancer_repo_test.go (hunk 2 above)
Comment (MartinForReal) [reply pkg/provider/azure_loadbalancer_repo_test.go]: I will rename it to azureconfig. Thanks!

File: pkg/provider/azure.go [hunk 4]
```diff
@@ -27,14 +27,13 @@ import (
 	"sync"
 	"time"
 
-	ratelimitconfig "sigs.k8s.io/cloud-provider-azure/pkg/provider/config"
+	azureconfig "sigs.k8s.io/cloud-provider-azure/pkg/provider/config"
```
Comment (nilo19) [inline pkg/provider/azure.go:30]: please sort the import

**Pull request closed without merging**